lib/openstax/auth/configuration.rb

```diff
@@ -2,11 +2,18 @@ module OpenStax
   module Auth
     class Configuration
 
-      # Settings for strategy 1
-
+      # settings for strategy 1
       attr_accessor :strategy_1_secret_key
       attr_accessor :strategy_1_secret_salt
       attr_accessor :strategy_1_cookie_name
+
+      # settings for strategy 2
+      attr_accessor :strategy_2_signature_public_key
+      attr_accessor :strategy_2_encryption_private_key
+      attr_accessor :strategy_2_cookie_name
+      attr_accessor :strategy_2_encryption_algorithm
+      attr_accessor :strategy_2_encryption_method
+      attr_accessor :strategy_2_signature_algorithm
```

Comment:
Since highlights is the only code that uses this gem so far, you could refactor here a bit and use `Strategy1` and `Strategy2` inner classes on the `Configuration` class.  Would save the repetition of `strategy_2_`

```ruby
class Configuration
  class Strategy1
    attr_accessor :secret_key
    ...
  end
 
  class Strategy2
    attr_accessor :signature_public_key
    ...
  end
end
```

would have to change the `Configuration` initializer to create and hold new instances of these inner classes which would need their own `attr_reader`.

Not a big deal either way, I yield to you on it.

Reply:
yeah, i like the idea of creating better organization in the configuration